### PR TITLE
Add EditDialog variant to Popover component

### LINKED ISSUE #1373: Add Edit Dialog pattern
# Do not use `EditDialog` within a full page form/record layout. This not a replacement for Inline Edit.

**It is meant for standalone "modify this word" parts of a page.**

This is not an Inline Edit pattern, but does have a similar UX pattern. If you were using an InlineEdit pattern outside a form, then this Edit Dialog variant of input is the replacement component. An Inline Edit UX pattern _within_ a form is A COMPLETELY DIFFERENT THING and is outside the scope of this Github issue. You should go look at #1395 

DSR's Inline Edit Form Input is a different pattern than https://www.lightningdesignsystem.com/components/form-element/#Inline-Edit

* Current `InlineEdit` component should be deprecated. [DONE]
* There is no close or cancel on blur with the new examples. ~Inline Edit appears to be upgrade path to [Fuel UX's Placard pattern](http://getfuelux.com/javascript.html#placard), but this should be discouraged.~ [DONE]
* ~Deprecation warning of current `InlineEdit` should point to a new `Popover` example on the doc site that shows an edit icon triggering a Popover with a base `Input` within it.~ [DONE]
* ~An example of a Inline Edit Form that swaps a read-only `Input` (that has a right-aligned edit icon on it) for a Base variant `Input` should be added to `Input`. Example should look like https://www.lightningdesignsystem.com/components/form-element/#View-Mode and https://www.lightningdesignsystem.com/components/form-element/#Edit-Mode (this is a different issue and outside scope of this issue, that should be considered a new Form component with a docked footer)~ #1395 

~We'll need to do this in consideration of https://github.com/salesforce/design-system-react/issues/273 which involves moving `components/form/InlineEdit` to `components/InlineEdit`, but in agreement with this issue should not be moved. It should be "removed" from the doc site and deprecated.~ [DONE]

**This is NOT a breaking change.**

### PR DESCRIPTION
**Pull request by davidlygagnon**
Fixes #1373 

### Additional description
- Added new component: `EditDialogPopover`, that brings up a popover for "inline edit" like interaction.
- `EditDialogPopover` defaults to having a `cancel` and `save` button
- Exposed two new props: `classNameBody` and `classNameFooter` on `popover` component to allow setting classNames on `popover_body` and `popover_footer` required from the design specs.
- 

Screen Shot 2019-04-25 at 11 05 10 PM

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by interactivellama** [comment]
Looking good! @davidlygagnon 

I think you might have just made @shaundr 's day. 🎉

**Comment by davidlygagnon** [comment]
Awesome! Let's ship it ! Just realizing there's a bunch of conflicts - will take a look today

**Comment by tanhengyeow** [comment]
@davidlygagnon I would be happy to help move forward this PR if you are occupied right now. It would also help me to learn from the work at the same time 😄

**Comment by davidlygagnon** [comment]
@tanhengyeow Thanks but please hold off on this one - I have a few things already planned that I'd like to do.

**Comment by tanhengyeow** [comment]
@davidlygagnon No problem! Always happy to help if it is needed :)

**Comment by interactivellama** [comment]
The spec isn't clear on this still, since we don't have this on the SLDS website.

**Comment by interactivellama** [comment]
It looks like Snyk is causing trouble that it didn't used to. I will look into it.

File: components/component-docs.json [hunk 1]
```diff
@@ -743,13 +743,6 @@
                     "computed": false
                 }
             },
-            "className": {
```
Comment (davidlygagnon) [inline components/component-docs.json]: Interestingly - maybe it's been a while we didn't run `build:docs` ? There's a bunch of updates unrelated to this PR.

**Comment by davidlygagnon** [comment]
@interactivellama PR's ready for review. Thanks!

**Comment by interactivellama** [comment]
@davidlygagnon Thank you for sticking this one out with all the back and forth. 🙇

**Pull request merged**